File: code/python/2501.py
# https://leetcode.com/problems/longest-square-streak-in-an-array/
class Solution:
    def longestSquareStreak(self, nums: List[int]) -> int:
        nums = sorted(list(set(nums)))
        calc = {}
        res = 0
        for num in nums:
            if num in calc:
                res = max(res, calc[num] + 1)
                calc[num * num] = calc[num] + 1
            else:
                calc[num * num] = 1
        return res if res > 1 else -1

File: code/cpp/2501.cpp
// https://leetcode.com/problems/longest-square-streak-in-an-array/
class Solution {
public:
    int longestSquareStreak(vector<int>& nums) {
        const size_t maxValue = 100001;
        bool flag[maxValue](false);
        for (int num : nums) flag[num] = true;
        int res = -1;
        for (size_t i = 0; i < 317; i++) {
            if (!flag[i]) continue;
            size_t idx = i;
            int count = 1;
            while ((idx = idx * idx) < maxValue and flag[idx]) {
                count++;
            }
            if (count > 1) res = max(res, count);
        }
        return res;
    }
};
